pynxtools_xps/scienta/scienta_txt.py

```diff
@@ -464,111 +458,158 @@ def _get_key_value_pair(self, line):
         """
         try:
             [key, value] = line.split("=")
+            key = convert_pascal_to_snake(key)
             key = self._re_map_keys(key)
             value = self._re_map_values(key, value)
-            return key, value
 
         except ValueError:
             key, value = "", ""
-            return key, value
 
-    def _re_map_keys(self, key):
+        return key, value
+
+    def _re_map_keys(self, key: str):
         """
         Map the keys returned from the file to the preferred keys for
         the parser output.
 
         """
-        maps = {}
-        for key_map in self.key_maps:
-            maps.update(key_map)
+        if key in KEY_MAP:
+            return KEY_MAP[key]
+        return key
 
-        keys = list(maps.keys())
+    def _re_map_values(self, input_key: str, value: str):
+        """
+        Map the values returned from the file to the preferred format for
+        the parser output.
 
-        if key in keys:
-            key = maps[key]
+        """
+        try:
+            value = value.rstrip("\n")
+        except AttributeError:
+            pass
 
-        return key
+        value_map = {
+            "no_of_regions": int,
+            "energy_size": int,
+            # "energy_axis",
+            "pass_energy": float,
+            "no_of_scans": int,
+            "excitation_energy": float,
+            "center_energy": float,
+            "start_energy": float,
+            "stop_energy": float,
+            "step_size": float,
+            "dwell_time": float,
+            "detector_first_x_channel": int,
+            "detector_last_x_channel": int,
+            "detector_first_y_channel": int,
+            "detector_last_y_channel": int,
+            "time_per_spectrum_channel": float,
+            "energy_units": _extract_energy_units,
+            "energy_axis": _separate_dimension_scale,
+            "energy_scale": _change_energy_scale,
+            "energy_scale_2": _change_energy_scale,
+            "acquisition_mode": _change_scan_mode,
+            "time_per_spectrum_channel": float,
+        }
 
-    def _change_no_of_regions(self, no_of_regions):
-        """
-        Make no_of_regions an integer.
-        Maps e.g. from '0001' string to 1.
+        keys = list(value_map.keys())
 
-        Parameters
-        ----------
-        no_of_regions : str
-            No. of regions in the data file,
-            read from the first line.
+        for k in keys:
+            if k in input_key:
+                map_method = value_map[k]
+                value = map_method(value)  # type: ignore[operator]
+        return value
 
-        Returns
-        -------
-        int
-            Ineger value of the number of regions.
 
-        """
-        return int(no_of_regions)
+def str_to_int(str_value: str):
```

Comment:
It's maybe a good idea to collect these functions in pynx.utils and use it from there as it is also useful for other readers. I think I already put some similar functions there. For now it's fine but I think generally it's good if we offer some standard helper functions in the main converter itself.

Reply:
I will collect them and then make a PR on pynxtools 👍